homeassistant/components/wled/__init__.py

```diff
@@ -115,26 +84,68 @@
     return True
 
 
+def wled_exception_handler(func):
+    """Decorate WLED calls to handle WLED exceptions.
+
+    A decorator that wraps the passed in function, catches WLED errors,
+    and handles the availability of the device in the data coordinator.
+    """
+
+    async def handler(self, *args, **kwargs):
+        try:
+            return await func(self, *args, **kwargs)
+        except WLEDError as error:
+            _LOGGER.error("Error communicating with API: %s", error)
+            self.coordinator.last_update_success = False
```

Comment:
Should this also trigger an update of the entities ?

Reply:
This approach also means that if we were to send invalid parameters in a turn on command to the WLED API, we mark all entities as unavailable ?

Reply:
This mimics the current behavior, but yes, the connection issue can be caught, as it throws a `WLEDConnectionError` when that happens (which is inherited from `WLEDError`).

So, on a connection error, mark all unavailable, else, just log the error? Of let the exception bubble up in case of the bad request?

Reply:
Oh, the first question: trigger an update of the entities: Yes, it should, however, there isn't a public method available in the DataUpdateCoordinator to signal the even listeners right now... (Unless I'm mistaken in my approach/train of thought).

Reply:
ok, so added a call to `await self.coordinator.async_refresh()` to it, which makes the on/off, brightness & effect things. It resolved jumpiness in the UI for those, however, for a color change, it made it worse.

The WLED has a transition on color changes, so scheduling an instant refresh makes the color dial jump when selecting colors...

So this feels a bit like I have to make the least bad choice.

Reply:
> there isn't a public method available in the DataUpdateCoordinator to signal the event listeners right now

You're extending it and so you can add it to the WLED Data Coordinator. Just add a test case for it and you're sure it will never break.

Reply:
done!